src/components/tasks/TaskSearch/TaskSearch.tsx

```diff
@@ -16,84 +23,316 @@ const initialSuggestionCoordinates: SuggestionCoordinates = {
+                            selectedFilters.length - 1
+                        );
+                    } else {
+                        const newOptions = selectedFilters.filter(
+                            (_, idx) => idx !== onRemoveSelectedFilterIndex
+                        );
+                        setSelectedFilters(newOptions);
+                        setOnEditSelectedFilterIndex(false);
+                        onClickSearchButton(
+                            convertSearchOptionsToQuery(newOptions)
+                        );
+                        setOnRemoveSelectedFilterIndex(-1);
+                    }
+                }
+
+                break;
+
+            case 'ArrowUp':
+                if (activeFilterSuggestionDropdownIndex > -1) {
+                    setActiveFilterSuggestionDropdownIndex(
+                        activeFilterSuggestionDropdownIndex - 1
+                    );
+                    event.preventDefault();
+                }
+                break;
+            case 'ArrowDown':
+                if (
+                    filterSuggestions.length - 1 >
+                    activeFilterSuggestionDropdownIndex
+                ) {
+                    setActiveFilterSuggestionDropdownIndex(
+                        activeFilterSuggestionDropdownIndex + 1
+                    );
+                    event.preventDefault();
+                }
+                break;
+            case 'Enter': {
+                if (activeFilterSuggestionDropdownIndex > -1) {
+                    onSuggestionSelected();
+                } else if (
+                    onEditSelectedFilterIndex !== false &&
+                    onEditSelectedFilterValue.length > 0 &&
+                    activeFilterSuggestionDropdownIndex !== -1
+                ) {
+                    onSuggestionSelected();
+                } else if (
+                    onEditSelectedFilterValue.length === 0 &&
+                    typedInput.length === 0
+                ) {
+                    setActiveFilterSuggestionDropdownIndex(-1);
                     searchButtonHandler();
                 }
                 break;
+            }
+            case 'Escape':
+                onEditSelectedFilterIndex !== false &&
+                    setOnEditSelectedFilterIndex(false);
+                break;
             default:
                 break;
         }
     };
 
+    const onResizeHandler = () => {
+        setSuggestionCoordinates(findCoordinates());
+    };
+    useEffect(onResizeHandler, [
+        defferedPillValue,
+        defferedUserInput,
+        filterSuggestions.length,
+    ]);
+    const removePill = (idx: number) => {
+        const updatedOptions = selectedFilters.filter(
+            (_, index) => index !== idx
+        );
+        setSelectedFilters(updatedOptions);
+        setOnEditSelectedFilterValue('');
+        toggleInputFocus();
+        onClickSearchButton(convertSearchOptionsToQuery(updatedOptions));
+    };
```

Comment:
`onResizeHandler` function, for example, is called whenever the window is resized. If this function contains complex or time-consuming operations, it could affect the performance of the application, especially if it's called frequently.


- Consider debouncing or throttling the function if it's performing expensive operations to prevent it from being called excessively.
- Ensure that any event listeners or subscriptions set up in this function are properly managed and cleaned up when they're no longer needed to avoid memory leaks.
- Review the `onResizeHandler` function one more time and assess if any operations within it can be optimized or if there are any unnecessary calculations.